### ISSUE TITLE
column resizing issue in force columns mode (no horizontal scroll) when container get smaller

### ISSUE THREAD

**Issue by barouchmaximedenoun** [swimlane/ngx-datatable#1686]
I find an issue in force mode column recalculation and I can also propose a fix for this issue. I will submit soon (tomorrow) a Pull Request to be reviewed.

The following example is to demonstrate the issue and is very little coding:
https://stackblitz.com/edit/angular-2e7nxq

In force columns mode with no horizontal scroll, when the table container is getting bigger (when I am hiding the slider clicking Show/Hide Slider button), I am sending a window resize event (in method triggerTableResize in app.component.ts) and the table resize its column perfectly accordingly to the new container width. But if I shrink the container (expanding the slider clicking a second time on Show/Hide Slider button) the table resize its column with a little miscalculation (the ellipsis button in the table is cut).

The problem is in ngx-datatable forceFillColumnWidths in utils.Math.ts,

When the remainingWidth is positiive we will go several time in 
do {
} while (remainingWidth > 0 && columnsToResize.length !== 0);

but when the container shrink and the remainingWidth is negative we just go once.

i will submit a Pull Request tomorrow with a fix.

resize the window which adapt the size of the column

**I'm submitting a ...**  (check one with "x")
```
[x ] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, post on Stackoverflow or Gitter
```

**Current behavior**
https://stackblitz.com/edit/angular-2e7nxq
clicking twice Show/Hide Slider button make the last column cut.
expand a bit the window the last column is not cut anymore.

**Expected behavior**
The last column is not cut.

**Reproduction of the problem**
https://stackblitz.com/edit/angular-2e7nxq
click twice on Show/Hide Slider button.
look at last column with ellipsis button which is cut

**What is the motivation / use case for changing the behavior?**

when the table container shrink because of a slider or going from full screen to normal screen  clciking on a button the last column will always be cut.

**Please tell us about your environment:**

reproductible on stackblitz.com also in my environment, Windows 10 Pro, Webstorm, npm
* **Table version:** 

14.0.0
* **Angular version:** 

v7

* **Browser:** [all | Chrome XX | Firefox XX | IE XX | Safari XX | Mobile Chrome XX | Android X.X Web Browser | iOS XX Safari | iOS XX UIWebView | iOS XX WKWebView ]

Chrome at least
* **Language:** [all | TypeScript X.X | ES6/7 | ES5]
all type script 3.1.1

**Comment by marjan-georgiev** [comment]
Merged the PR. This will be released in 15.1.0

**Comment by barouchmaximedenoun** [comment]
Great Thank you very much
